### PR TITLE
chore(trie): `lib/trie/hash.go` tests

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

- Near full coverage and deeper tests for lib/trie/hash.go
- Small production code changes

## Tests

```
go test -race github.com/ChainSafe/gossamer/lib/trie
```

## Issues

- #1973

### CODE REVIEW

File: lib/trie/hash_test.go [hunk 1]
```diff
@@ -4,72 +4,1009 @@
...
+			} else {
+				require.NoError(t, err)
+			}
+		})
 	}
 }
 
-func TestHashBranch(t *testing.T) {
-	n := &branch{key: generateRandBytes(380), value: generateRandBytes(380)}
-	n.children[3] = &leaf{key: generateRandBytes(380), value: generateRandBytes(380)}
```
Comment (noot) [inline lib/trie/hash_test.go:47]: should we add a test for random nodes as well?

**Review by noot — APPROVED** [review]
looks good to me! how do you feel about adding more tests using `generateRand` haha

**Review by edwardmack — APPROVED** [review]
Nice work, lgtm!

File: lib/trie/hash_test.go (hunk 1 above)
Comment (qdm12) [reply lib/trie/hash_test.go:47]: Why? :thinking: 
I'm a big fan of deterministic tests.
Unless we need crypto randomness in certain niche cases (i.e.generate uniform distribution etc.)?

**Pull request merged**